code/modules/projectiles/ammunition/magazines.dm

```diff
@@ -178,7 +178,10 @@
 	ammo_type = /obj/item/ammo_casing/c10mm
 	caliber = "10mm"
 	max_ammo = 8
-	multiple_sprites = 2
+
+/obj/item/ammo_box/magazine/m10mm/update_icon()
+	..()
+	icon_state = "[initial(icon_state)]-[stored_ammo.len ? "8" : "0"]"
```

Comment:
Make the `update_icon` for the subtype, since `multiple_sprites = 2` is just fine for default magazines.